estuary-cdk/estuary_cdk/http.py

```diff
@@ -237,7 +238,9 @@ async def _request_stream(
             await asyncio.sleep(cur_delay)
 
             headers = {}
-            if _with_token and self.token_source is not None:
+            if self.auth_headers:
```

Comment:
I had made a comment here, but I think it got lost somehow (I must have forgot to add it): I think we should modify `TokenSource` to have an optional `authorization_header` parameter that defaults to `Authorization`, but could allow for an alternate header to be used in cases like this. That would be more consistent with the usage of `TokenSource`.